pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -221,40 +212,8 @@ func kubeBinaryURL(kubernetesVersion, cpuArch string) string {
 	return fmt.Sprintf("%s/kubernetes/v%s/binaries/kubernetes-node-linux-%s.tar.gz", globalAKSMirror, kubernetesVersion, cpuArch)
 }
 
-// Check and initialize each field if it is nil
-func initializeIfNil[T any](field **T) {
-	if *field == nil {
-		*field = new(T)
-	}
-}
-
-// EnsureConfigsNonNil checks if the config is nil and initializes it if it is.
-func ensureConfigsNonNil(v *nbcontractv1.Configuration) *nbcontractv1.Configuration {
-	if v == nil {
-		v = &nbcontractv1.Configuration{}
-	}
-
-	initializeIfNil(&v.KubeBinaryConfig)
-	initializeIfNil(&v.ApiServerConfig)
-	initializeIfNil(&v.AuthConfig)
-	initializeIfNil(&v.ClusterConfig)
-	initializeIfNil(&v.NetworkConfig)
-	initializeIfNil(&v.GpuConfig)
-	initializeIfNil(&v.TlsBootstrappingConfig)
-	initializeIfNil(&v.KubeletConfig)
-	initializeIfNil(&v.RuncConfig)
-	initializeIfNil(&v.ContainerdConfig)
-	initializeIfNil(&v.TeleportConfig)
-	initializeIfNil(&v.CustomLinuxOsConfig)
-	initializeIfNil(&v.HttpProxyConfig)
-	initializeIfNil(&v.CustomCloudConfig)
-	initializeIfNil(&v.CustomSearchDomainConfig)
-
-	return v
-}
-
 func (a AKS) applyOptions(v *nbcontractv1.Configuration) {
-	nbcontract := ensureConfigsNonNil(v)
+	nbcontract := nbcontracthelper.NewNBContractConfiguration(v)
```

Comment:
Given what this function does and how it is used, I think it would be better to treat it like a constuctor that does not take any input and returns an initialized contract. And so, the bootstrapper would just call `nbcontracthelper.NewNBContractConfiguration()` to get an instance of the contract. Then, i think we can create a separate function like `applyStaticBootstrapVars`, instead of declaring `staticNodeBootstrapVars`. What do you think?

Reply:
Agreed. Updated.